### PR TITLE
ENH: support sparse X for glmnet

### PR DESCRIPTION
**Pull request by mathurinm**
I added rcv1.binary to the datasets, it seems to work, and quite fast, with the policy of #46 

image

However on finance I get a huge memory consumption, + 18 Gb:
image

### CODE REVIEW

**Review by agramfort — APPROVED** [review]

**Comment by mathurinm** [comment]
@jolars can I get your opinion on this ?

**Comment by jolars** [comment]
LGTM!

image

Regarding memory consumption, I'm not really sure what's going on, but you are making at least one additional copy so maybe that explains it at least partly. How large is the difference compared to fitting through python?

**Review by tomMoral — APPROVED** [review]
LGTM! Thx @mathurinm

File: objective.py [hunk 1]
```diff
@@ -8,8 +8,8 @@ class Objective(BaseObjective):
     name = "Lasso Regression"
 
     parameters = {
-        'reg': [0.05, .1, .5],
-        'fit_intercept': [True, False]
+        'fit_intercept': [True, False],
+        'reg': [.5, .1, .05],
     }
```
Comment (tomMoral) [inline objective.py:13]: Not sure why this change was made but this broke the documentation on `benchopt` as an implicit assumption was made on the order of the parameters.

`benchopt/benchopt#265` fixes this by ensuring a deterministic order for the parameters.

Comment (mathurinm) [reply objective.py:13]: Thanks ! Random changed happened after tweaking objective configurations for debug

Comment (tomMoral) [reply objective.py:13]: As it was unsafe, I guess this is good as this is now fixed ;)
